command/agent/http.go

```diff
@@ -366,7 +366,13 @@ func (s *HTTPServer) ResolveToken(req *http.Request) (*acl.ACL, error) {
 	var err error
 
 	if srv := s.agent.Server(); srv != nil {
-		aclObj, err = srv.ResolveToken(secret)
+		r := &structs.GenericRequest{}
+		r.AuthToken = secret
+		if authErr := srv.Authenticate(nil, r); authErr != nil {
+			return nil, fmt.Errorf("ACL token not found or invalid workload identity: %v", authErr)
```

Comment:
In server endpoints in the `nomad` package these are always `ErrPermissionDenied`. We should return them unwrapped.